src/HeaderList.php

```diff
@@ -25,7 +33,11 @@ public function __construct(array $names)
      */
     public static function fromString($string)
     {
-        return new static(explode(' ', $string));
+        if (is_null($string)) {
+            return ['date'];
```

Comment:
Likewise here, seems like it might be the wrong place for this type of logic? Either should live in the code where it is necessary, or we should create a method to encapsulate this "exception" behaviour

Reply:
Resolved in https://github.com/99designs/http-signatures-php/pull/42/commits/5a5928c35fde1918a6f3320de3ed98f7845f65c4